vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -159,49 +153,76 @@ public Set<T> getSelectedItems() {
                 .unmodifiableSet(new LinkedHashSet<>(selected.values()));
     }
 
+    /**
+     * Returns an unmodifiable view of the selected item ids.
+     *
+     * The returned Set may be a direct view of the internal data structures of this class.
+     * A defensive copy should be made by callers when iterating over this Set and modifying
+     * the selection during iteration to avoid ConcurrentModificationExceptions.
+     */
+    protected Set<Object> getSelectedItemIds() {
+        return Collections.unmodifiableSet(this.selected.keySet());
```

Comment:
Maybe it makes sense to add this to `SelectionModel` interface and so having this as a 
```@Override
    public Set<Object> getSelectedItemIds() {
        return Collections.unmodifiableSet(this.selected.keySet());
       }```
here. 

 In that case, you also need to implement this method in `GridNoneSelectionModel`, but I think it would be a solution overall.

Reply:
It seems like @web-padawan has more to say about your API's (as in the other comment). Care to weigh in here as well ?